src/python/pants/backend/native/rules/targets.py

```diff
@@ -0,0 +1,230 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from typing import Dict, Iterable, Optional, Tuple, Union
+
+from pants.backend.native.subsystems.native_build_step import ToolchainVariant
+from pants.backend.native.targets.external_native_library import ConanRequirement
+from pants.backend.native.targets.native_artifact import NativeArtifact
+from pants.build_graph.address import Address
+from pants.engine.target import (
+    COMMON_TARGET_FIELDS,
+    BoolField,
+    Dependencies,
+    InvalidFieldException,
+    InvalidFieldTypeException,
+    PrimitiveField,
+    Sources,
+    StringField,
+    StringSequenceField,
+    Target,
+)
+from pants.util.collections import ensure_list, ensure_str_list
+from pants.util.frozendict import FrozenDict
+
+
+class CSources(Sources):
+    default = ("*.h", "*.c")
+
+
+class CppSources(Sources):
+    default = ("*.h", "*.hpp", "*.cpp")
+
+
+class CtypesNativeLibrary(PrimitiveField):
+    alias = "ctypes_native_library"
+    value: NativeArtifact
+    required = True
+
+    @classmethod
+    def compute_value(
+        cls, raw_value: Optional[NativeArtifact], *, address: Address
+    ) -> NativeArtifact:
+        value = super().compute_value(raw_value, address=address)
+        if not isinstance(value, NativeArtifact):
+            raise InvalidFieldTypeException(
+                address, cls.alias, value, expected_type="a `native_artifact` object"
+            )
+        return value
+
+
+class FatalWarnings(BoolField):
+    alias = "fatal_warnings"
+    default = False
+
+
+class StrictDeps(BoolField):
+    """Whether to include only dependencies directly declared in the BUILD file.
+
+    If this is False, all transitive dependencies are used when compiling and linking native code.
+    """
+
+    alias = "strict_deps"
+    default = False
+
+
+class ToolchainVariantField(StringField):
```

Comment:
I started adding an `EnumField` base class but gave up because I realized that we need to store the `value` as a `str` rather than `ToolchainVariant` for now so that we can safely pass a string to the V1 Target constructor.